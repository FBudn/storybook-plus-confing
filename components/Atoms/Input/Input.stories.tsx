import React from "react";
import { View } from "react-native";
import Input from "./Input";

const InputMeta = {
  title: "Atoms/Input",
  component: Input,
  argTypes: {
    setInputValue: { action: "Input value changed " },
  },
  args: {
    testId: "test-input-id",
    secureText: "boolean",
  },
  decorators: [
    (Story: any) => (
      <View
        style={{
          alignItems: "stretch",
          justifyContent: "center",
          flex: 1,
          backgroundColor: "white",
        }}
      >
        <Story />
      </View>
    ),
  ],
};

export default InputMeta;

export const Basic = {};
