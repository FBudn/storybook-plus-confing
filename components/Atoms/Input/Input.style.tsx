import { StyleSheet } from "react-native";

const InputStyles = StyleSheet.create({
  container: {
    borderRadius: 5,
    borderWidth: 2,
    borderStyle: "solid",
    borderColor: "rgb(214 211 209)",
    height: 40,
    width: "100%",
  },
});

export default InputStyles;
