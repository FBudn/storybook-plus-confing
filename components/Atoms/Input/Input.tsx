import React from "react";
import { TextInput, View } from "react-native";
import InputStyles from "./Input.style";

export interface InputProps {
  testId: string;
  secureText: boolean;
}

const Input: React.FC<InputProps> = ({ testId, secureText }) => (
  <View>
    <TextInput
      style={InputStyles.container}
      data-testid={testId}
      testID={testId}
      secureTextEntry={secureText}
    />
  </View>
);

export default Input;
