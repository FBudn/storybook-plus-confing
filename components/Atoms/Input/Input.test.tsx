/* eslint-disable @typescript-eslint/no-unnecessary-type-assertion */
import React from "react";
import { fireEvent, render, screen } from "@testing-library/react-native";
import TestRenderer from "react-test-renderer";
import Input, { InputProps } from "./Input";

const mockInputFunction = jest.fn();
const props: InputProps = {
  testId: `test-input-id`,
  secureText: true,
  setInputValue: mockInputFunction,
};

test(`Should render Input component`, () => {
  render(<Input {...props} />);
  const InputElement = screen.getByTestId(`test-input-id`);
  expect(InputElement).toBeOnTheScreen();
});

test(`Should match snapshot`, () => {
  const InputElement = TestRenderer.create(<Input {...props} />).toJSON();
  expect(InputElement).toMatchSnapshot();
});

test(`Should render and have styles:`, () => {
  render(<Input {...props} />);
  const InputElement = screen.getByTestId(`test-input-id`);
  expect(InputElement).toBeOnTheScreen();

  expect(InputElement).toHaveStyle({
    borderRadius: 5,
    borderWidth: 2,
    borderStyle: "solid",
    borderColor: "rgb(214 211 209)",
    height: 40,
  });
});

test(`Should render Input and handle setInputValue`, () => {
  render(<Input {...props} />);
  const InputElement = screen.getByTestId(`test-input-id`);
  expect(InputElement).toBeOnTheScreen();
  expect(mockInputFunction).toBeCalledTimes(0);
  fireEvent.press(InputElement);
  expect(mockInputFunction).toBeCalledTimes(1);
});
