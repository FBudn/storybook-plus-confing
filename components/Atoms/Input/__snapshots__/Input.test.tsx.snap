// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<TextInput
  data-testid="test-input-id"
  placeholder="jestem"
  style={
    {
      "borderColor": "rgb(214 211 209)",
      "borderRadius": 5,
      "borderStyle": "solid",
      "borderWidth": 2,
      "height": 40,
      "width": "100%",
    }
  }
  testID="test-input-id"
/>
`;
